source/Serial/USBSerial.cpp

```diff
@@ -209,6 +209,17 @@ UsbSerial::flush(
     )
 {
     _current_store_operation = _tx->StoreAsync();
+    create_task( _current_store_operation )
+        .then( [this]( task<unsigned int> task_ )
+    {
+        //detect disconnection
```

Comment:
Can StoreAsync() fail? If so, you should be calling task_.get() under a try block to catch the exception. If not, you don't need a task-based continuation here.